test/unit/ids.js

```diff
@@ -16,26 +17,31 @@ const mockDb = (idFilterLogicFn) => {
 };
 
 exports.tearDown = callback => {
-    testUtils.restore([
-        Math.random
-    ]);
+  testUtils.restore([
+    Math.random
+  ]);
 
-    callback();
+  callback();
 };
 
-exports['generates an id of the given length'] = function(test) {
-    [1,2,3,4,5,6,7,8,9,10].forEach(function(l) {
-        test.equal(ids._generate(l).length, l);
-    });
+exports['generates an id of the given length'] = test => {
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach(l =>
+    test.equal(ids._generate(l).length, l));
 
-    test.done();
+  test.done();
 };
 
-exports['ids can start with 0, will be correct length'] = function(test) {
-    sinon.stub(Math, 'random').returns(0.00001);
+exports['ids can start with 0, will be correct length'] = test => {
+  sinon.stub(Math, 'random').returns(0.00001);
 
-    test.equal(ids._generate(5), '00000');
-    test.done();
+  test.equal(ids._generate(5), '00000');
+  test.done();
+};
+
+exports['ids are "always" the length they should be'] = test => {
+  test.ok(
+    jsc.checkForall(jsc.integer(5, 13), i => ids._generate(i).length === i));
```

Comment:
Hey thanks for this explanation, unfortunately I still don't understand.  I probably need one more example or something.  Anyway didn't mean for you to spend too much time on it.  Basically I'm completely lost in the paragraph below, probably because I have no experience with QuickCheck.

"In the jsverify / quickcheck / property based check / generative test (first two are libraries, second two are names for basically the same thing, for your later googling) I am declaring a property that this algorithm has. It is then up to the library to exercise and properly test that library. This test should catch the specific edge case I know about, but it could catch others as well."